dev/reference/tools/data-migration/usage-scenarios/best-practice-dm-shard.md

```diff
@@ -0,0 +1,142 @@
+---
+title: Best Practice of Data Migration in the Shard Merge Scenario
+summary: Learn the best practice of data migration in the shard merge scenario.
+category: reference
+---
+# Best Practice of Data Migration in the Shard Merge Scenario
+
+This document details the supported functions and limitations of TiDB Data Migration (DM) in the shard merge scenario and provides a best practice guide to facilitate application for users.
+
+## An independent data migration task
+
+In [Merge and Replicate Data from Sharded Tables](/dev/reference/tools/data-migration/features/shard-merge.md#principles), the concept of sharding group is introduced. Simply put, a sharding group consists of all upstream tables that need to be merged and replicated into a same downstream table.
+
+The current sharding DDL mechanism has some [usage restrictions](/dev/reference/tools/data-migration/features/shard-merge.md#restrictions) to coordinate the schema changes using DDL statements in different sharded tables. If these restrictions are found to be violated due to unexpected reasons, you need to [handle sharding DDL locks manually in DM](/dev/reference/tools/data-migration/features/manually-handling-sharding-ddl-locks.md), or even redo the entire data migration.
+
+Therefore, to mitigate the impact of exception handling on data migration, it is recommended to merge and replicate each sharding group in separate data migration task. *In this case, when an exception is thrown, only a small number of data migration tasks might need to be handled manually while others remain unaffected.*
+
+## Handle sharding DDL locks manually
+
+In [Merge and Replicate Data from Sharded Tables](/dev/reference/tools/data-migration/features/shard-merge.md#principles), it is clear that sharding DDL lock in DM is a mechanism for coordinating the execution of DDL statements from multiple upstream sharded tables.
+
+Therefore, when you confirm there are sharding DDL locks on the DM-master through `show-ddl-locks` command, or some DM-workers have `unresolvedGroups` or `blockingDDLs` through `query-status` command, do not rush to unlock the sharding DDL lock manually using `unlock-ddl-lock` or `break-ddllock`.
+
+Only when you confirm that the current failure to automatically unlock the sharding DDL lock is listed in the [supported scenarios](/dev/reference/tools/data-migration/features/manually-handling-sharding-ddl-locks.md#supported-scenarios), can you follow the corresponding solution. For unsupported scenarios, it is recommended to redo the entire data migration. That is, redo the full and incremental data migration after emptying the data in the downstream database and the `dm_meta` information associated with the migration task.
```

Comment:
```suggestion
Instead, you can:

- Follow the corresponding manual solution to handle the scenario if the failure of automatically releasing the sharding DDL lock is one of the [listed abnormal scenarios](/dev/reference/tools/data-migration/features/manually-handling-sharding-ddl-locks.md#supported-scenarios).
- Redo the entire data migration task if it is an unsupported scenario: First, empty the data in the downstream database and the `dm_meta` information associated with the migration task; then, re-execute the full and incremental data migration.
```
Note: try not to use inverted sentences in our technical documents.

Reply:
Got it!